src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -503,4 +507,104 @@ public void replaceVariablesInFile(String filePath, List<String> targetStrings,
         }
         Files.write(replaceFilePath, fileContent.getBytes(charset));
     }
+
+    /**
+     * This normalizes all line endings to UNIX-line-endings recursively from the startPath.
+     *
+     * @see {@link #normalizeLineEndings(String) normalizeLineEndings}
+     * @param startPath          the path where the start directory is located
+     * @throws IOException if an issue occurs on file access for the normalizing of the line endings.
+     */
+    public void normalizeLineEndingsRecursive(String startPath) throws IOException {
+        log.debug("Normalizing file endings in directory {}", startPath);
+        File directory = new File(startPath);
+        if (!directory.exists() || !directory.isDirectory()) {
+            throw new RuntimeException("File endings in directory " + startPath + " should be normalized but the directory does not exist.");
+        }
+
+        // Get all files in directory
+        Collection<File> files = FileUtils.listFiles(directory, FileFilterUtils.trueFileFilter(), FileFilterUtils.trueFileFilter());
+
+        for (File file : files) {
+            normalizeLineEndings(file.getAbsolutePath());
+        }
+    }
+
+    /**
+     * This normalizes all line endings to UNIX-line-endings in a specific file.
+     * '\r\n' gets replaced to '\n'
+     * '\r' gets replaced to '\n'
+     *
+     * @param filePath           the path where the file is located
+     * @throws IOException if an issue occurs on file access for the normalizing of the line endings.
+     */
+    public void normalizeLineEndings(String filePath) throws IOException {
+        log.debug("Normalizing line endings in file {}", filePath);
+        // https://stackoverflow.com/questions/3776923/how-can-i-normalize-the-eol-character-in-java
+        Path replaceFilePath = Paths.get(filePath);
+        Charset charset = StandardCharsets.UTF_8;
+
+        String fileContent = new String(Files.readAllBytes(replaceFilePath), charset);
+        fileContent = fileContent.replaceAll("\\r\\n?", "\n");
+        Files.write(replaceFilePath, fileContent.getBytes(charset));
+    }
+
+    /**
+     * This converts all files to the UTF-8 encoding recursively from the startPath.
+     *
+     * @see {@link #convertToUTF8(String) convertToUTF8}
+     * @param startPath          the path where the start directory is located
+     * @throws IOException if an issue occurs on file access when converting to UTF-8.
+     */
+    public void convertToUTF8Recursive(String startPath) throws IOException {
```

Comment:
I think the name is misleading, as this implementation does not utilize recursion.

Reply:
This was a remnant of an earlier implementation, I changed it to be more specific.